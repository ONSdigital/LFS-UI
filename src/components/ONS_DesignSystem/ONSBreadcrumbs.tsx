import React, {Component} from "react";
import {Link} from "react-router-dom";

interface Props {
    List: any
    Current: string
}

export class ONSBreadcrumbs extends Component <Props, {}> {
    render() {
        return(
            <nav className="breadcrumb" aria-label="Breadcrumb">
                <ol className="breadcrumb__items">
                    {this.props.List.map((item:any, index: number) => (
                        <li key={index} id={"breadcrumb-" + (index + 1)} className="breadcrumb__item u-fs-s">
                            <Link to={"/" + item.link} className="breadcrumb__link">{item.name}</Link>
                        </li>
                    ))}
                    
                    <li id={"breadcrumb-" + String(this.props.List.length + 1) } className="breadcrumb__item u-fs-s breadcrumb__item--current">
                        {this.props.Current}
                    </li>
                </ol>
            </nav>
        )
    }
}