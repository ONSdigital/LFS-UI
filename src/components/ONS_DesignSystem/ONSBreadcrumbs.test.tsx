import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import {ONSBreadcrumbs} from "./ONSBreadcrumbs";
import Adapter from "enzyme-adapter-react-16";
import {BrowserRouter} from "react-router-dom";

describe("ONS Breadcrumbs Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const Props = {
        List: [],
        Current: "The page you are on"
    }

    const listProps = {
        List: [{name: "1", link: ""}],
        Current: "The page you are on"
    }

    function wrapper(render: any, props: any) {
        return render(
            <BrowserRouter>
                <ONSBreadcrumbs
                    {...props}>
                </ONSBreadcrumbs>
            </BrowserRouter>
        )
    }

    it("matches Snapshot", () => {
        expect(wrapper(shallow, Props)).toMatchSnapshot()
    });

    it("should render correctly", () => expect(wrapper(shallow, Props).exists()).toEqual(true));

    it("should render the correct current text", () => {
        expect(wrapper(mount, listProps).find("ONSBreadcrumbs").getElement().props.Current).toEqual(listProps.Current)
    })

})