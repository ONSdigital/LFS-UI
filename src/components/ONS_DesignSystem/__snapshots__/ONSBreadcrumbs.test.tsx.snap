// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Breadcrumbs Test matches Snapshot 1`] = `ShallowWrapper {}`;
